statemachine/statemachine.go

```diff
@@ -319,6 +393,21 @@ func run(f StatefulHandler, tid string, cmd <-chan Message) (m Message) {
 // Stop sends a Release message to the state machine through the command chan.
 func (s *stateMachine) Stop() {
 	s.cmds <- Message{Code: Release}
+
+	// Also inform the state machine it should exit since the internal handler
+	// may override the release message causing the task to be unreleaseable.
+	s.stop()
+}
+
+func (s *stateMachine) stop() {
+	s.stopL.Lock()
+	defer s.stopL.Unlock()
+	select {
+	case <-s.stopped:
+		return
+	default:
+		close(s.stopped)
```

Comment:
should this be wrapped in a sync.Once() so we dont' close it more than once?   *that may not be possible, so ignore as needed.